R/teal_slice.R

```diff
@@ -268,22 +280,68 @@ c.teal_slice <- function(...) {
   ans
 }
 
+#' @param x `teal_slice` object
+#' @keywords internal
+to_list <- function(x){
+  checkmate::assert_class(x, "teal_slice")
+
+  x <- if (shiny::isRunning()) {
+    shiny::reactiveValuesToList(x)
+  } else {
+    shiny::isolate(shiny::reactiveValuesToList(x))
+  }
+
+  formals_names <- setdiff(names(formals(filter_var)), '...')
+
+  extra_names   <- setdiff(names(x), formals_names)
+  missing_names <- setdiff(formals_names, names(x))
+
+
+  x <- c(x, formals(filter_var)[missing_names])
+
+  x[c(formals_names, extra_names)]
+
+}
+
+#' @param x `list` object
+#' @param ... further arguments passed to [jsonlite::toJSON]
+#' @keywords internal
+to_json <- function(x, ...) {
+  checkmate::assert_class(x, "list")
+  jsonlite::toJSON(x, pretty = TRUE, auto_unbox = TRUE, digits = 16, ...)
+}
 
 # format method for teal_slice
 #' @export
+#' @param x `teal_slice` object
+#' @param show_all `logical(1)` should parameters set to NULL be returned
 #' @rdname teal_slice
 #' @keywords internal
 #'
-format.teal_slice <- function(x, show_all = FALSE, ...) {
+format.teal_slice <- function(x, show_all = FALSE, pretty = TRUE, ...) {
   checkmate::assert_flag(show_all)
+  checkmate::assert_flag(pretty)
 
-  x <- if (shiny::isRunning()) {
-    rev(shiny::reactiveValuesToList(x))
-  } else {
-    rev(shiny::isolate(shiny::reactiveValuesToList(x)))
-  }
+  x_json <- to_json(to_list(x))
+  x_json_c <- capture.output(print(x_json))
+
+  if (!show_all) x_json_c <- grep('{}', x_json_c, fixed = TRUE, invert = TRUE, value = TRUE)
+  if (!pretty) return(x_json_c)
+
+  x_p_json <- pretty_json(x_json_c)
+
+  paste(c("teal_slice", x_p_json), collapse = "\n")
+}
+
+# centering of json output for `teal_slices` object JSON representation
+#' @param x a result of `to_json(to_list(teal_slice))`
+#' @keywords internal
+#'
+pretty_json <- function(x){
```

Comment:
```suggestion
pretty_json <- function(x) {
```

Reply:
@chlebowa I was thinking about the name. What would you say about `center_json` instead of `pretty_json`?

Reply:
9c57aa627d3e099b51651f1093914b566b6559ae